homeassistant/components/sensor/airvisual.py

```diff
@@ -95,60 +94,63 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     if city and state and country:
         _LOGGER.debug(
             "Using city, state, and country: %s, %s, %s", city, state, country)
+        location_id = city + state + country
```

Comment:
Separate with comma: `','.join((city, state, country))

Reply:
Compared to what I have, this seems to produce a different hash each time (causing multiple duplicates in the entity registry).

Reply:
Is your hashing deterministic? 😉 

This should generate a new string of `city, state, country`

Reply:
So strange! I'll keep digging.

Reply:
Well, should have researched this one: turns out that [since Python 3.3, `hash()` is randomized and not deterministic](https://jimmycallin.github.io/2016/01/03/make-your-python3-code-reproducible/). Thankfully, I'm not going insane. 😆Going to push a different checksum algorithm shortly.